WordPress/Classes/ViewRelated/Notifications/Views/NoteBlockImageTableViewCell.swift

```diff
@@ -5,20 +5,25 @@ import WordPressShared.WPStyleGuide
 class NoteBlockImageTableViewCell: NoteBlockTableViewCell
 {
     // MARK: - Public Properties
+    private var imageURL: NSURL?
     override var isBadge: Bool {
         didSet {
             backgroundColor = isBadge ? Styles.badgeBackgroundColor : Styles.blockBackgroundColor
         }
     }
 
     // MARK: - Public Methods
-    func downloadImageWithURL(url: NSURL?) {
-        let success = { (image: UIImage) in
-            self.blockImageView.image = image
-            self.blockImageView.displayWithSpringAnimation()
+    func downloadImageIfNeededWithURL(url: NSURL?) {
```

Comment:
Instead of changing the method name, why not adding documentation that explains the methods behavior?

The `ifNecessary` part makes the reader wonder what "necessary" means, but is not really clarifying what to expect, whereas documentation would be pretty specific about it.